mobile/cmp/button/ColChooserButton.js

```diff
@@ -23,13 +23,16 @@ export class ColChooserButton extends Component {
 
     static propTypes = {
         /** GridModel of the grid for which this button should show a chooser. */
-        model: PT.instanceOf(GridModel).isRequired
+        model: PT.instanceOf(GridModel).isRequired,
+
+        /** Allow pinning the first row **/
+        enablePinFirstRow: PT.bool
```

Comment:
The issue in my mind is that we came up with the idea of supporting this as a special flag for mobile only, where it seemed to make sense to have a simpler model for pinning that could be controlled / viewed from the updated column chooser.

I find it difficult to evaluate and balance out competing demands of flexibility / simplicity / consistency between the mobile and desktop toolkits for features like this.

Maybe it would help to enumerate a few things to structure my thoughts...

### What pinned states do we envision being of use?

1. No columns pinned.
2. One column pinned to the left.
    - Most common case I think by a good margin.
3. Multiple columns pinned to the left.
    - Can imagine relatively common case where two columns are pinned - e.g. some kind of minimal status / category column + a primary label / identifier column.
4. Column(s) pinned to the left and right.
    - Do not believe right pinned columns would be common. I can imagine uses for them, but in the real world you do not usually see this.

### How could users pin a column / columns?

1. Use the drag gesture built-in to ag-grid, dragging a column in the grid to either the left or right until the little "pin" icon shows up in the draggable.
    - 👍 Built-in to ag-grid.  Allows quick pinning of any number of columns to either left or right.
    - 👎 Potentially problematic on mobile where dragging can be tricky or less reliable, especially on the problematic browsers we sometimes encounter.
    - 👎 Not easily discoverable for users who don't know about the feature.
    - 👎 Pinned state wouldn't be represented in column chooser without also doing some work there.
2. Toggle a simplified option in the column chooser to pin the first column, covering the most common pinning case (2) above.
    - 👍 Can be presented as a simple, easily understood option to users within the column chooser.
    - 👎 Not compatible with multiple pinned columns, or right pinned columns.
3. Toggle a control on each displayed column in the column chooser turning pinning on/off for that column. 
    - 👍 Could represent pinned state of multiple columns on both left and right.
    - 👍 Could be used alongside dragging option (1).
    - 👎 Most complex to implement. To be clear and usable, would need to account for column order - i.e. what should happen if you could request a column in the middle of the list be pinned? What should happen when you re-order columns?

My motivation for suggesting (2) on mobile as an incremental first step was to minimize complexity and handle the case I think we really need now on mobile, without relying on the drag-to-pin gesture. But maybe it's an uncanny valley that just sets us up for more work in the near future and we should push forwards to (3).